### DISCUSSION TITLE
Does ansible-lint include a check for the new FQCNs?

### DISCUSSION THREAD

**Discussion by loganmarchione** [ansible/ansible-lint#1061] [Q&A] [answered]
##### Summary
As the title says, does ansible-lint include a check to replace all my old module names with the new [FQCNs](https://docs.ansible.com/ansible/latest/user_guide/collections_using.html#using-collections-in-a-playbook)?

I know I can use `-vv` when I actually run the playbook, but if there was an ansible-lint check, that would be easier.

##### Issue Type

- Feature Idea

##### Additional Information

N/A

**Comment by mdlopresti** [comment]
This would be nice for integrating a FQCN check into CI process.

**Reply by ssbarnea** [reply]
No, it does not but when https://github.com/ansible-community/ansible-lint/pull/950 will be merged, we will be able to rely on ansible to resolve FQCN names.

**Comment by loganmarchione** [comment, marked as answer]
Closing this out, it seems it does now, thanks!

https://ansible-lint.readthedocs.io/en/latest/default_rules.html?highlight=fqcn#fqcn-builtins

**Reply by GhostLyrics** [reply]
Neither 5.1.2 nor 5.1.3 shipped with the FQCN rule for me when I tried to display them with `-L` despite the docs saying it does so.

**Reply by loganmarchione** [reply]
@GhostLyrics  - I had to put this in my `.ansible-lint` file.
```
enable_list:
  - fqcn-builtins  # opt-in
```

Now I see alerts like below in my CI/CD pipeline.
```
fqcn-builtins: Use FQCN for builtin actions
roles/install_iac_tools/tasks/main.yml:26 Task/Handler: Unarchive
```

**Reply by GhostLyrics** [reply]
Thank you, using `enable_list` for opt-ins solved my issue.

**Reply by bluikko** [reply]
Adding `enable_list` gives me a schema validation error in `ansible-lint.json`. I believe this is a feature of the YAML extension for VSCode which I have not configured in any particular way for this kind of use. It seems to have a default setting to use https://www.schemastore.org/api/json/catalog.json for detecting schemas. Maybe the schema is out of date somewhere? Any ideas?

Seems to use https://raw.githubusercontent.com/ansible-community/schemas/main/f/ansible-lint.json which indeed doesn't list `enable_list`.
